pyomo/core/base/set.py

```diff
@@ -1409,7 +1410,19 @@ def add(self, *values):
 
             if self._validate is not None:
                 try:
-                    flag = self._validate(_block, _value)
+                    # differentiate between indexed and non-indexed sets
+                    if self._index is not None:
+                        # indexed set: the value and the index are given
+                        pass
```

Comment:
I do not think this `pass` is necessary.

Reply:
True. Not sure how it landed there. Will fix it soon.